src/runtime_src/core/edge/user/aie/aie.cpp

```diff
@@ -120,6 +178,20 @@ open_context(const xrt_core::device* device, xrt::aie::access_mode am)
   access_mode = am;
 }
 
+void
+Aie::
+open_context(const xrt_core::device* device, const zynqaie::hwctx_object* hwctx_obj, xrt::aie::access_mode am)
+{
+  auto drv = ZYNQ::shim::handleCheck(device->get_device_handle());
+
+  //TODO: replace openAIEContext with new function with parameters hwctx_obj, am
+  int ret = drv->openAIEContext(am);
+  if (ret)
+    throw xrt_core::error(ret, "Fail to open AIE context");
```

Comment:
Nit.  Consider same in constructor.
```suggestion
  if (auto ret = drv->openAIEContext(am))
    throw xrt_core::error(ret, "Fail to open AIE context");
```